### PR TITLE
Add ability to change log level per deployment

### PR DESCRIPTION
**Pull request by jakobs**

### CODE REVIEW

File: lib/orocos/process.rb [hunk 1]
```diff
@@ -790,6 +811,14 @@ def spawn(options = Hash.new)
                     ENV['LD_PRELOAD'] = Orocos.tracing_library_path
                 end
 
+                if options[:log_level]
+                    if [:debug, :info, :warn, :error, :fatal].include? options[:log_level]
+                        ENV['BASE_LOG_LEVEL'] = options[:log_level].to_s.upcase
+                    else
+                        Orocos.warn "'#{options[:log_level]}' is not a valid log level."
```
Comment (doudou) [inline lib/orocos/process.rb]: I would prefer raising here. Warnings get lost.

**Comment by jakobs** [comment]
raising now instead of warn.

File: lib/orocos/process.rb [hunk 2]
```diff
@@ -783,6 +804,16 @@ def spawn(options = Hash.new)
...
             end
             cmdline = [module_bin]
+
+            # check arguments for log_level
+            log_level = nil
+            if options[:log_level]
+                if [:debug, :info, :warn, :error, :fatal, :disable].include? options[:log_level]
+                    log_level = options[:log_level].to_s.upcase
+                else
+                    raise ArgumentError, "'#{options[:log_level]}' is not a valid log level."
```
Comment (doudou) [inline lib/orocos/process.rb]: Display the list of valid log levels in the error message, that makes for a much more useful message.

**Comment by jakobs** [comment]
improved the error message

**Comment by doudou** [comment]
Merged.

**Pull request merged**